### ISSUE TITLE
Best weights

### ISSUE THREAD

**Issue by soft-sensor** [lightly-ai/lightly-train#235]
How do I save best weights? not just last but best, and is it based on train loss or what other metrics option do we have to save best weights?

**Comment by stegmuel** [comment]
For methods that do not use labels (e.g., `distillation` or `dinov2`), there is no reliable way to assess a model’s downstream performance at a given epoch without actually evaluating it. Even shallow evaluations, such as k-NN classification, can be misleading.
Therefore, our current recommendation is to train multiple models using increasingly long schedules and run the full evaluation on each of them.

**Comment by soft-sensor** [comment]
but what about train loss? can't we save on the basis of train loss itself?

**Comment by stegmuel** [comment]
Unfortunately, the training loss isn’t very informative for this purpose.
For example, the DINOv2 training curves below show successful runs, yet it’s difficult to pinpoint any training step that could clearly be considered better than the others.
<img width="1709" height="766" alt="Image" src="https://github.com/user-attachments/assets/4a6abc77-9840-4c3b-9739-74687ece3a6a" />

In distillation, the training loss generally decreases monotonically, but there’s definitely a point beyond which continuing to train no longer brings improvements.

A more practical approach than what I suggested above might be to simply save models at different iterations (see this [issue](https://github.com/lightly-ai/lightly-train/issues/225)) and then evaluate them.
